File: hw/elbay/u1/hw07Collections/Collections.js
class Person {
    constructor(id, fullName, bornOnDate, alive) {
        this.id = id;
        this.fullName = fullName;
        this.bornOnDate = bornOnDate;
        this.alive = alive;
    }

    computeAge() {
        let today = new Date();

        let age = today.getFullYear() - this.bornOnDate.getFullYear();

        let monthDifference = today.getMonth() - this.bornOnDate.getMonth();

        if (
            monthDifference < 0 ||
            (monthDifference === 0 &&
                today.getDate() < this.bornOnDate.getDate())
        ) {
            age--;
        }

        return age;
    }

    toString() {
        return `Person{id=${this.id}, fullName='${this.fullName}', bornOnDate=${this.bornOnDate.toDateString()}, alive=${this.alive}}`;
    }
}


let id;
let fullName;
let bornOnDate;
let alive;


// Persona 1
id = 1;
fullName = "Didier Elbay";
bornOnDate = new Date(1970, 11, 17); // Mes empieza en 0
alive = true;

let person = new Person(id, fullName, bornOnDate, alive);

console.log(person.toString());
console.log("Age of person 1 is " + person.computeAge());


// Persona 2
id = 2;
fullName = "Tarjelia Tello";
bornOnDate = new Date(1910, 7, 8);
alive = false;

let person2 = new Person(id, fullName, bornOnDate, alive);

console.log("Age of person 2 is " + person2.computeAge());


// Lista things
let things = [];

things.push(person);
things.push(person2);

console.log("Size of things is " + things.length);


// Lista integers
let integers = [];

integers.push(5);
integers.push(id);

console.log("integers:", integers);


// Lista people
let people = [];

people.push(person);
people.push(person2);

console.log("people:", people);

File: hw/elbay/u1/hw07Collections/Collections.py
from datetime import date

class Person:
    def __init__(self, id, full_name, born_on_date, alive):
        self.id = id
        self.full_name = full_name
        self.born_on_date = born_on_date
        self.alive = alive

    def compute_age(self):
        today = date.today()

        age = today.year - self.born_on_date.year

        if (today.month, today.day) < (self.born_on_date.month, self.born_on_date.day):
            age -= 1

        return age

    def __str__(self):
        return f"Person(id={self.id}, full_name='{self.full_name}', born_on_date={self.born_on_date}, alive={self.alive})"


def main():

    id = 1
    full_name = "Didier Elbay"
    born_on_date = date(1970, 12, 17)
    alive = True

    person = Person(id, full_name, born_on_date, alive)

    print(person)
    print("Age of person 1 is", person.compute_age())

    
    id = 2
    full_name = "Tarjelia Tello"
    born_on_date = date(1910, 8, 8)
    alive = False

    person2 = Person(id, full_name, born_on_date, alive)

    print("Age of person 2 is", person2.compute_age())

    
    things = []
    things.append(person)
    things.append(person2)

    print("Size of things is", len(things))

    
    integers = []
    integers.append(5)
    integers.append(id)

    print("integers:", integers)

    
    people = []
    people.append(person)
    people.append(person2)

    print("people:", people)


if __name__ == "__main__":
    main()